### PR TITLE
Fix file /api/search endpoint when collections not configured as a postgres view.

### PR DESCRIPTION
**Pull request by ewelinajozwik**
Fixes FAIRSPC-106

### CODE REVIEW

File: projects/saturn/src/main/java/io/fairspace/saturn/services/search/JdbcFileSearchService.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+@Log4j2
+public class JdbcFileSearchService implements FileSearchService {
+    private final Transactions transactions;
+    private final CollectionResource rootSubject;
+    private final Config.Search searchConfig;
+    private final ViewsConfig viewsConfig;
+    private final ViewStoreClientFactory viewStoreClientFactory;
+
+    public JdbcFileSearchService(
```
Comment (tgreenwood) [inline projects/saturn/src/main/java/io/fairspace/saturn/services/search/JdbcFileSearchService.java:26]: you can use @RequiredArgsContructor by lombok instead

File: projects/saturn/src/main/java/io/fairspace/saturn/services/search/JdbcFileSearchService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Transactions transactions,
+            CollectionResource rootSubject) {
+        this.searchConfig = searchConfig;
+        this.viewStoreClientFactory = viewStoreClientFactory;
+        this.transactions = transactions;
+        this.rootSubject = rootSubject;
+        this.viewsConfig = viewsConfig;
+    }
+
+    @SneakyThrows
```
Comment (tgreenwood) [inline projects/saturn/src/main/java/io/fairspace/saturn/services/search/JdbcFileSearchService.java:39]: why do you use SneakyThrows? Also, try block does not catch anything, I would rather catch there, log it and re-throw unchecked exception if needed

Comment (ewelinajozwik) [reply projects/saturn/src/main/java/io/fairspace/saturn/services/search/JdbcFileSearchService.java:39]: For this change I focused only on fixing the bug, which required restructuring the code. The code however is the same as it was (I only copy-pasted it to a different class). I didn't focus on the improvements - SneakyThrows was just there. If it needs to be changed, more classes needs to be refactored, same for the RequiredArgsContructor - for consistency we should change it everywhere then.

**Review by tgreenwood — APPROVED** [review]
No need to improve in the scope of this ticket for sure, missed that it is just copy/paste. In this scope all look good to me

**Pull request merged**